Sources/GRPCCore/Call/Server/ServerContext.swift

```diff
@@ -45,6 +45,15 @@ public struct ServerContext: Sendable {
   /// - "in-process:27182".
   public var localPeer: String
 
+  /// An optional field for transports to store specific data
+  ///
+  /// Refer to the transport documentation to understand what type of
+  /// value this field will contain, if any.
+  ///
+  /// An example of what this field can be used for, would be to store
+  /// things like a peerCertificate from a mTLS connection
+  public var transportSpecific: (any Sendable)?
```

Comment:
I thought about this some more, rather than `any Sendable` can you define a `public protocol TransportSpecific: Sendable {}` nested within `ServerContext` and use that instead?

It has a few advantages:
1. Compiler helps users out a little more
2. It's easier to document and easier for users to find implementations.